pkg/registries/registries.go

```diff
@@ -20,6 +20,15 @@ func scopeMatchesRegistry(scope, reg string) bool {
 	return false
 }
 
+// isSubScopeOfReg returns true if scope is a value that is a sub-scope of reg
+// e.g *.foo.example.com is a sub-scope of *.example.com or bar.example.com/bar is a sub-scope of *.example.com
+func isSubScopeOfReg(scope, reg string) bool {
+	if strings.HasPrefix(reg, "*.") {
+		return strings.HasSuffix(scope, reg[1:]) || strings.Contains(scope, reg[1:]+"/") || strings.Contains(scope, reg[1:]+":")
```

Comment:
`*.foo` matches `bar/baz.foo`, `bar/baz.foo/x`, `bar/baz.foo:500`, and shouldn’t: extract host[:port] from `scope` before matching, or verify that the part matched by `*` doesn’t contain a `/` (the former is probably simpler).

Reply:
A little clarification, should wild card entries not match any host[:port] entries? I am a little confused of the matching issue here.

Reply:
Most importantly the (domain) wildcard should not match namespace / repo name.

WRT requiring a port match for a wildcard, I’ll need to read a bit more.

Reply:
Got it, fixed and added test cases.

Reply:
(Mostly note to self, for the record): The c/image semantics is that `*.com` does match `example.com:5000/foo`.